src/map.rs

```diff
@@ -14,29 +14,47 @@ pub struct MapBundle {
     pub transform: Transform,
     pub global_transform: GlobalTransform,
 }
-
-/// TODO: DOCS
-pub struct Map {
+pub struct MapSettings {
     pub size: MapVec2,
     pub chunk_size: MapVec2,
     pub tile_size: Vec2,
     pub texture_size: Vec2,
     pub layer_id: u32,
-    chunks: HashMap<MapVec2, (Entity, Vec<Option<Entity>>)>,
     pub mesher: Box<dyn TilemapChunkMesher>,
+}
+
+impl Clone for MapSettings {
+    fn clone(&self) -> Self {
+        Self {
+            size: self.size.clone(),
```

Comment:
I think these `Vec2`s should be `Copy` instead of `.clone()` the compiler would optimize this on release, but if I'm not mistaken, wouldn't optimize it on debug.

also, is it possible to rename `size` to `map_size`?